### PR TITLE
Typescript generator

### LINKED ISSUE #130: [SDK] Typescript generator
### GOAL:
Similar to how DAODAO handles types, configure typescript output for use in web/typescript projects.

### Requirements:
- [x] See: https://github.com/DA0-DA0/dao-contracts/tree/main/typescript
- [x] Setup new "typescript" folder with similar setup to DAODAO
- [x] Add to CI/CD flow, so types get generated & built into ts upon new versions

### PR DESCRIPTION
**Pull request by deveusss**
closes #130

### CODE REVIEW

**Comment by mikedotexe** [comment]
Thanks for this! Should we remove the `types` folder?

File: typescript/.gitignore [hunk 1]
```diff
@@ -0,0 +1,21 @@
+
+# macOS
+.DS_Store
```
Comment (mikedotexe) [inline typescript/.gitignore:3]: No need to change this, but in the future, I believe we can use project-level `.gitignore` files to be specific to application. Oftentimes it makes sense to use a "global" `.gitignore` for files like `.DS_Store` and others.

Check out this guide:
https://sebastiandedeyne.com/setting-up-a-global-gitignore-file

**Review by mikedotexe — APPROVED** [review]
hey @deveusss can you check #196 real quick? but besides that stamping this approval, thanks!

**Comment by deveusss** [comment]
@mikedotexe thank you, merged)

**Pull request merged**